onadata/apps/api/tools.py

```diff
@@ -360,7 +354,11 @@ def id_string_exists_in_account():
                 xform.save()
         except IntegrityError:
             raise exceptions.ParseError(_(msg))
-        set_project_perms_to_xform_async.delay(xform.pk, project.pk)
+        try:
+            set_project_perms_to_xform_async.delay(xform.pk, project.pk)
+        except librabbitmq.ConnectionError:
+            # Apply permissions synchrounously
+            set_project_perms_to_xform(xform, project)
```

Comment:
No, we cannot connect to the broker, we have no access to a queue in order to queue the background retry. 

We have had the most complaints about permissions on form publishing and sharing, in most cases applying in them in the foreground as a backup will be sufficient. For those with a lot of users who have access to the project, it might take a little bit longer and the client might drop the connection but the permissions will still be applied. Trying to avoid the situation where we have to schedule a cron job to reapply permission every x minutes/hours.